test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,195 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 				return kcCmNamesCur
 			}).WithContext(ctx).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Equal(kcCmNamesPre))
 		})
+
+		It("[reboot_required][slow] should be able to correctly identify topology manager policy without scheduler restarting", Label("reboot_required", "slow"), func() {
+			nroOperObj := &nropv1.NUMAResourcesOperator{}
+			nroKey := objects.NROObjectKey()
+			err := fxt.Client.Get(context.TODO(), nroKey, nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+
+			mcps, err := nropmcp.GetListByNodeGroupsV1(context.TODO(), fxt.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred(), "cannot get MCPs associated with NUMAResourcesOperator %q", nroOperObj.Name)
+
+			kcList := &machineconfigv1.KubeletConfigList{}
+			err = fxt.Client.List(context.TODO(), kcList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var targetedKC *machineconfigv1.KubeletConfig
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						targetedKC = kc
+					}
+				}
+			}
+			Expect(targetedKC).ToNot(BeNil(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+			initialTopologyManagerPolicy := kcObj.TopologyManagerPolicy
+
+			By("modifying Topology Manager Policy under kubeletconfig to none")
+			Eventually(func(g Gomega) {
+				err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(targetedKC), targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				applyNewTopologyManagerPolicy(&kcObj.TopologyManagerPolicy, v1beta1.NoneTopologyManagerPolicy)
```

Comment:
let's add a comment why to change to "none" specifically or it can be other policy?

Reply:
also let's logout the current policy and if not single-numa-node let's report a warning , and if the same as `none` then break with a klog.Info log

Reply:
It can be any other policy but making it none and then changing to single-numa-node and then creating a tae deployment is the best option in my opinion to check this scenario, but it can be any policy though